runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -294,218 +269,258 @@ pub mod pallet {
+	> {
+		#[cfg(feature = "runtime-metrics")]
+		sp_io::init_tracing();
+
 		let ParachainsInherentData {
-			bitfields: mut signed_bitfields,
+			mut bitfields,
 			mut backed_candidates,
 			parent_header,
 			mut disputes,
 		} = data;
-		#[cfg(feature = "runtime-metrics")]
-		sp_io::init_tracing();
 
 		log::debug!(
 			target: LOG_TARGET,
-			"[enter_inner] parent_header={:?} bitfields.len(): {}, backed_candidates.len(): {}, disputes.len(): {}",
-			parent_header.hash(),
-			signed_bitfields.len(),
+			"[process_inherent_data] bitfields.len(): {}, backed_candidates.len(): {}, disputes.len() {}",
+			bitfields.len(),
 			backed_candidates.len(),
 			disputes.len()
 		);
 
-		// Check that the submitted parent header indeed corresponds to the previous block hash.
 		let parent_hash = <frame_system::Pallet<T>>::parent_hash();
+
 		ensure!(
 			parent_header.hash().as_ref() == parent_hash.as_ref(),
 			Error::<T>::InvalidParentHeader,
 		);
 
 		let now = <frame_system::Pallet<T>>::block_number();
 
-		let mut candidates_weight = backed_candidates_weight::<T>(&backed_candidates);
-		let mut bitfields_weight = signed_bitfields_weight::<T>(signed_bitfields.len());
+		let candidates_weight = backed_candidates_weight::<T>(&backed_candidates);
+		let bitfields_weight = signed_bitfields_weight::<T>(bitfields.len());
 		let disputes_weight = multi_dispute_statement_sets_weight::<T, _, _>(&disputes);
 
-		let current_session = <shared::Pallet<T>>::session_index();
-
-		let max_block_weight = <T as frame_system::Config>::BlockWeights::get().max_block;
-
 		METRICS
 			.on_before_filter((candidates_weight + bitfields_weight + disputes_weight).ref_time());
 
-		T::DisputesHandler::assure_deduplicated_and_sorted(&mut disputes)
-			.map_err(|_e| Error::<T>::DisputeStatementsUnsortedOrDuplicates)?;
-
-		let (checked_disputes, total_consumed_weight) = {
-			// Obtain config params..
-			let config = <configuration::Pallet<T>>::config();
-			let post_conclusion_acceptance_period =
-				config.dispute_post_conclusion_acceptance_period;
+		let current_session = <shared::Pallet<T>>::session_index();
+		let expected_bits = <scheduler::Pallet<T>>::availability_cores().len();
+		let validator_public = shared::Pallet::<T>::active_validator_keys();
+		let max_block_weight = <T as frame_system::Config>::BlockWeights::get().max_block;
 
-			let verify_dispute_sigs = if let FullCheck::Yes = full_check {
-				VerifyDisputeSignatures::Yes
-			} else {
-				VerifyDisputeSignatures::Skip
-			};
+		let entropy = compute_entropy::<T>(parent_hash);
+		let mut rng = rand_chacha::ChaChaRng::from_seed(entropy.into());
 
-			// .. and prepare a helper closure.
-			let dispute_set_validity_check = move |set| {
-				T::DisputesHandler::filter_dispute_data(
-					set,
-					post_conclusion_acceptance_period,
-					verify_dispute_sigs,
-				)
-			};
+		// Filter out duplicates and continue.
+		if let Err(()) = T::DisputesHandler::deduplicate_and_sort_dispute_data(&mut disputes) {
```

Comment:
Can the input here be large when the chain is loaded with disputes ? Are we covering up elsewhere for the weight used here ?

Reply:
Yes weight is covered again via benchmarks of the whole enter call. For create_inherent the provisioner has to limit the amount to something reasonable (e.g. to not overflow the wasm memory), but that code is in place as well.